### DISCUSSION TITLE
Release naming and older versions

### DISCUSSION THREAD

**Discussion by wmcnCG** [OpenRadioss/OpenRadioss#2044] [Ideas]
Is there a way to access older releases? Why are they being removed? 

I use a script to setup new machines with OpenRadioss and I often have to update the download link because the release is no longer available (this would be solved if the most recent release was always available at https://github.com/OpenRadioss/OpenRadioss/releases/download/latest/OpenRadioss_linux64.zip). 

Besides this minor inconvenience, I think older releases should remain available for reproducibility.

**Comment by PaulAltair** [comment]
The suggestion of having a fixed URL for the 'latest' packaged version is a good one in my opinion, I will suggest to see if we can do that. Regarding retaining 'older' versions, OpenRadioss is effectively versionless, every pull request updates the code and there is no real specific identifier for a 'release', retaining old versions would mean keeping a copy of the download for every pull request. If you want to retain a 'version' for use later, or across multiple machines. You can obviously reuse the packaged version you downloaded. Or if you create your own fork, it remains at the state in which you forked it, until you specifically sync it back up with the main repository updates.

**Reply by PaulAltair** [reply]
Any 'previous' build of OpenRadioss can be retrieved and recompiled by using `$ git checkout <commit-id>` if you want to get an older version where the package is no longer hosted as an exe/binary

You can view the log of commits with `$ git log'` and return to current build with `$git checkout main`

![checkout_on_given_commit_1](https://github.com/OpenRadioss/OpenRadioss/assets/98418895/5549aeb9-eb93-467b-b903-35699e7527e8)
![Checkout_on_commit_4c6147](https://github.com/OpenRadioss/OpenRadioss/assets/98418895/65b7f29b-f150-4c1a-b952-b6dabc13fc8d)

**Comment by mquinzin** [comment]
Hi to easily find the latest download URLs, you can use API like this for example:

`curl -s https://api.github.com/repos/OpenRadioss/OpenRadioss/releases/latest  | grep browser_download_url
      "browser_download_url": "https://github.com/OpenRadioss/OpenRadioss/releases/download/latest-20231109/OpenRadioss_linux64.zip"
      "browser_download_url": "https://github.com/OpenRadioss/OpenRadioss/releases/download/latest-20231109/OpenRadioss_win64.zip"`
